packages/core/src/sync/sync-api.ts

```diff
@@ -262,11 +266,16 @@ export class SyncApi implements ISyncApi {
       fromBlock: startBlock,
       toBlock: endBlock,
       models: modelIds,
+      syncCompleteCallback: this.modelSyncComplete,
     })
   }
 
+  modelSyncComplete(model: string) {
+    this.modelsToHistoricSync.delete(model)
```

Comment:
a history sync can occur in the follow situations:
1. When a node is restart and needs to catch up
2. A node is syncing for the first time
3. A reorg happened and we need to re-sync X blocks
4. We add a new model

a sync is complete for a model when it does not have a sync running due to 2 or 4. I am not sure about 1 or 3.  A model can have multiple history syncs at once. Ex. A node restarts twice so it has two different catchup syncs. 

So basically I don't think modelsToHistoricSync should be a set, rather than a mapping with counts

Reply:
That's actually a really interesting question.  I think I agree with blocking cases 2 and 4 but letting queries go through in cases 1 and 3.  @oed @3avi @michaelsena for any other opinions.